packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -550,14 +558,32 @@ class _StretchDragControllerProvider extends InheritedWidget {
 ///     `CupertinoSheetRoute`, with optional nested navigation built in.
 class CupertinoSheetRoute<T> extends PageRoute<T> with _CupertinoSheetRouteTransitionMixin<T> {
   /// Creates a page route that displays an iOS styled sheet.
-  CupertinoSheetRoute({super.settings, required this.builder, this.enableDrag = true});
+  CupertinoSheetRoute({
+    super.settings,
+    required this.builder,
+    this.enableDrag = true,
+    double? topGap,
+  }) : assert(
+         topGap == null || (topGap >= 0.0 && topGap <= 1.0),
+         'topGap must be between 0.0 and 1.0',
+       ),
+       _topGap = topGap;
 
   /// Builds the primary contents of the sheet route.
   final WidgetBuilder builder;
 
   @override
   final bool enableDrag;
 
+  // The gap between the top of the screen and the top of the sheet.
+  final double? _topGap;
+
+  @override
+  double get topGap => _topGap ?? _kTopGapRatio;
+
+  @override
+  bool get hasCustomTopGap => _topGap != null;
```

Comment:
This maybe should be a private method. I'm not sure if it's useful outside of these classes.